components/places/src/storage/bookmarks.rs

```diff
@@ -907,13 +907,9 @@ fn get_raw_bookmarks_for_url(db: &PlacesDb, url: &Url) -> Result<Vec<RawBookmark
 }
 
 fn reset_in_tx(db: &PlacesDb, assoc: &EngineSyncAssociation) -> Result<()> {
-    // Remove all synced bookmarks and pending tombstones, and mark all
-    // local bookmarks as new.
+    // Mark all local bookmarks as new so that we re-upload them on the next sync
     db.execute_batch(&format!(
-        "DELETE FROM moz_bookmarks_synced;
-
-        DELETE FROM moz_bookmarks_deleted;
```

Comment:
question: How come we want to carry over unknown fields between resets? Wouldn't re-downloading all records on the first sync after a reset also download those unknown fields?

Reply:
One argument for keeping them is just that we might be the first device to show up after a reset, right?

But although I haven't looked much at this patch, it does seem a red-flag that a reset isn't killing the mirror.

Reply:
(I haven't put as much thought into resets as I should. Eg, I don't think autofill will keep them because it relies on what's stored in the mirror and deletes it on reset). It would seem odd for some engines to carry them and not others though.)

Reply:
Maybe we need to define what a reset represents.  If reset means "forget the server data", then I think it should also mean that we forget the unknown fields.  If it means "reset your timestamps/counters so you do a full sync next time", then I think it's okay to keep the mirror data around.

Reply:
It means "we have disconnected from the server", either explicitly due to user action, or because the server state changed (eg, password reset). In the former case, we may never reconnect, so keeping old server data around forever just smells wrong from a privacy POV - their DB still having bookmarks deleted years sounds quite bad. In a server-state change, then we are going to reconnect soon, but that differentiation doesn't currently exist. I guess we could consider looking into adding it.